src/oop/static_modifier/medium/q5/MainTest.java

```diff
@@ -0,0 +1,114 @@
+        String[] methodModifiers = {"public"};
+        Object[][] arguments = {
+                {new String[0], String[].class}
+        };
+        outputClass.callMethod("main", arguments, methodModifiers, classInstance, new Clause[]{
+                new StringLiteral("This is a " + initialName + " that is " + initialColour + " in colour"),
+                new Optional(new StringLiteral(" ")),
+                new Optional(new NewLine())
+        });
+    }
+
+    @Test
+    @Order(2)
+    public void hatClassHasCorrectAttributes() {
+        String incorrectNameError = "Your %s class does not include the %s attribute.";
+        String incorrectVisibilityError = "Your %s attribute does not have the correct visibility modifier.";
+        String incorrectStaticError = "Your %s attribute does not have the static modifier.";
+        assertTrue(testClass.hasField(staticAttributeName, String.class),
+                String.format(incorrectNameError, className, staticAttributeName));
+        assertTrue(testClass.hasModifier(staticAttributeName, "private"),
+                String.format(incorrectVisibilityError, staticAttributeName));
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"),
+                String.format(incorrectStaticError, staticAttributeName));
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("Black", "Nike"),
+                Arguments.of("Blue", "Adidas"),
+                Arguments.of("Red", "Roots"),
+                Arguments.of("Brown", "Gap")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent1")
+    @Order(3)
+    public void correctToStringMethod(String colour, String compName) throws Throwable {
+        Object[][] arguments = {
+                {colour, String.class}
+        };
+        testClass.setFieldValue(null, compName, staticAttributeName);
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod("toString", null), String.format(incorrectMethodDefinition, "toString", className));
+        assertTrue(testClass.hasModifier("toString", null, "public"), String.format(incorrectModifierMessage, "toString", className));
+        assertTrue(testClass.hasReturnType("toString", null, String.class), String.format(incorrectReturnType, "toString", className));
+        String[] methodModifiers = {"public"};
+        String expected = "This is a " + compName + " that is " + colour + " in colour";
```

Comment:
You should use a formatted string and put this at the top of the class so we can vary it on the website.